src/auxinfo/participant.rs

```diff
@@ -527,7 +527,12 @@ impl AuxInfoParticipant {
 
         Ok((
             decryption_key.into(),
-            AuxInfoPublic::new(self.retrieve_context(), self.id(), encryption_key, params)?,
+            AuxInfoPublic::new(
+                self.retrieve_context(),
+                <AuxInfoParticipant as ProtocolParticipant>::id(self),
```

Comment:
todo: Now that there's only one `id` function, you can replace this with just `self.id()` -- you don't have to specify which trait it belongs to any more.